peachjam_search/views.py

```diff
@@ -134,14 +146,20 @@ class SearchFilterBackend(CompoundSearchFilterBackend):
     3. Combined simple and advanced, using both SHOULD and MUST from above.
     """
 
-    must_backends = [MultiFieldSearchQueryBackend()]
+    def __init__(self):
+        super().__init__()
 
-    should_backends = [
-        # Search each field individually using SimpleQueryString which allows quotes, +foo, -bar etc.
-        CrossFieldSimpleQueryStringBackend,
-        # Customised search on PDF page content
-        NestedPageQueryBackend,
-    ]
+        self.must_backends = [MultiFieldSearchQueryBackend()]
+
+        if pj_settings().pagerank_boost_value:
+            self.must_backends += [RankFeatureBackend()]
```

Comment:
Ah, I see. I think hard-code in the backends, and then the backend can return an empty list instead? Less fiddly than doing this here.

Reply:
That way, the backend is the only thing that cares about its config. Otherwise both this and the backend must care.

Reply:
changed :heavy_check_mark: